pkg/sloop/server/internal/config/config.go

```diff
@@ -129,23 +129,29 @@ func registerFlags(fs *flag.FlagSet, config *SloopConfig) {
 //
 // We do this to support settings that can come from either cmd line or config file
 func Init() *SloopConfig {
-	newConfig := &SloopConfig{}
-
-	configFilename := preParseConfigFlag()
-	glog.Infof("Config flag: %s", configFilename)
-	if configFilename == "" {
-		configFilename = os.Getenv(sloopConfigEnvVar)
-		glog.Infof("Config env: %s", configFilename)
+	finalConfig := &SloopConfig{}
+	registerDefaultFlags(flag.CommandLine, finalConfig)
+
+	configFileName:=""
+	configFileFlag := preParseConfigFlag()
+	configFileOS := os.Getenv(sloopConfigEnvVar)
+
+	if configFileFlag != "" {
+		configFileName = configFileFlag
+		glog.Infof("Config flag: %s", configFileFlag)
+	} else if configFileOS!=""{
+		configFileName=configFileOS
+		glog.Infof("Config env: %s", configFileOS)
 	}
-	if configFilename != "" {
-		newConfig = loadFromFile(configFilename)
-	} else {
-		registerFlags(flag.CommandLine, newConfig)
-		flag.Parse()
+
+	if configFileName != "" {
+		finalConfig = loadFromFile(configFileName,finalConfig)
 	}
+	//register cmd line args
```

Comment:
what would happen if configFileFlag == "" and configFileOS =="" and configFileName == ""

Reply:
it would use the default values.

Reply:
My concern would be all of the 3 argus are all "", how are we going to handle it, maybe we can have a log to print out these variables, it can help for debugging purposes.

Reply:
see L133, i register default flags. it will never be empty. We are already printing out the values in server.go.

Reply:
i see